client/src/components/profileInfoEdit/ProfileInfoEdit.js

```diff
@@ -18,31 +18,70 @@ import { updateTeam } from '../../actions/team';
 const ProfileInfoEdit = () => {
   const dispatch = useDispatch();
 
-  const { teamId, teamName, orgName, email } = useSelector(
+  const { teamId, teamName, orgName, email,password, confirmedPassword } = useSelector(
     (state) => state.team
   );
 
-  const [profileData, setInputs] = useState({ teamName, orgName, email });
+  const [profileData, setInputs] = useState({ teamName, orgName, email,password ,confirmedPassword});
 
   useEffect(() => {
-    setInputs({ teamName, orgName, email });
-  }, [email, orgName, teamName]);
+    setInputs({ teamName, orgName, email,password,confirmedPassword  });
+  }, [email, orgName, teamName,password,confirmedPassword]);
 
   const updateInputs = (form) => {
     const { name, value } = form.target;
-    setInputs({ ...profileData, [name]: value });
+    if (name !== "email" ){
+      setInputs({ ...profileData, [name]: value });
+    }
   };
+  const checkPassword = function(){
 
-  const [validated, setValidated] = useState(false);
-  const handleUpdate = (event) => {
-    const form = event.currentTarget;
-    event.preventDefault();
-    if (form.checkValidity() === false) {
-      event.stopPropagation();
-    } else {
-      dispatch(updateTeam(teamId, profileData));
+    if (! ({...profileData}.password === {...profileData}.confirmedPassword)){
+      return false;
     }
-    setValidated(true);
+    var matches = {...profileData}.password.match(
+        /^(?=.*[A-Za-z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/);
+    if (matches === null){
+      console.log("false");
+      return false;
+    }
+    return true;
+
+  };
+
+  const [validated, setValidated] = useState(false);
+  const handleUpdate = async (event) => {
+      const form = event.currentTarget;
+      event.preventDefault();
+      if (form.checkValidity() === false) {
+          event.stopPropagation();
+      } else {
+          var newdata;
+          if (({...profileData}.password || {...profileData}.confirmedPassword ) && ({...profileData}.password !== "" || {...profileData}.confirmedPassword !== "")) {
```

Comment:
For consistency - might be worth using formik/yup for form validation here since there's other parts of the frontend that use it like the register page.

Reply:
i tried that, but wasnt able to integrate it with the rest of the form on this page, this should be ok for now?